eth/protocols/eth/protocol.go

```diff
@@ -242,6 +277,20 @@ func (p *BlockBodiesPacket) Unpack() ([][]*types.Transaction, [][]*types.Header)
 	return txset, uncleset
 }
 
+// Unpack retrieves the transactions, uncles, blobs, proofs, and commitments from the range packet and returns
+// them in a split flat format that's more consistent with the internal data structures.
+func (p *BlockBodiesPacket100) Unpack() ([][]*types.Transaction, [][]*types.Header, [][]*types.BlobTxSidecar) {
+	var (
+		txset      = make([][]*types.Transaction, len((*p).BlockBodiesPacket))
+		uncleset   = make([][]*types.Header, len((*p).BlockBodiesPacket))
+		sidecarset = make([][]*types.BlobTxSidecar, len((*p).BlockBodiesPacket))
+	)
+	for i, body := range (*p).BlockBodiesPacket {
+		txset[i], uncleset[i], sidecarset[i] = body.Transactions, body.Uncles, (*p).Sidecars[i]
+	}
+	return txset, uncleset, sidecarset
+}
+
```

Comment:
Can we use p.BlockBodiesPacket instead of (*p).BlockBodiesPacket and some other places in this function?

Reply:
fixed, I saw the previous code used p since the pointer can't be used directly, but I didn't know pointers can be dereferenced automatically if called to its attributes